sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -2007,4 +2007,76 @@ class RegexUnsupportedException(message: String, index: Option[Int])
       case _ => message
     }
   }
-}
\ No newline at end of file
+}
+
+sealed trait RegexOptimizationType
+object RegexOptimizationType {
+  case class StartsWith(pattern: String) extends RegexOptimizationType
```

Comment:
minor nit:it's not a pattern, just a literal string wrapped in these case classes, correct?

Reply:
done.